app/models/candidates/school.rb

```diff
@@ -1,80 +1,20 @@
-module Candidates
-  class School
-    include ActiveModel::Model
-
-    DISTANCES = [
-      [1, '1 mile'],
-      [3, '3 miles'],
-      [5, '5 miles'],
-      [10, '10 miles'],
-      [15, '15 miles'],
-      [20, '20 miles'],
-      [25, '25 miles']
-    ].freeze
-
-    PHASES = [
-      ['4-11', 'Primary 4 - 11'],
-      ['11-16', 'Secondary 11 - 16'],
-      ['16-18', '16 to 18']
-    ].freeze
-
-    FEES = [
-      %w{none None},
-      ['<30', 'up to £30'],
-      ['<60', 'up to £60'],
-      ['<90', 'up to £90']
-    ].freeze
-
-    SUBJECTS = [
-      'Art and design', 'Combined science', 'Computer science',
-      'Food preparation and nutrition', 'Geography', 'History', 'Music',
-      'Physical education', 'Religious studies'
-    ].freeze
-
-    attr_accessor :query
-    attr_writer :distance
-    attr_accessor :fees, :subject, :phase
-
-    def distance
-      @distance.to_i
-    end
-
-    def results
-      school = Struct.new(:name, :address, :phase, :fees, :school_type, :subjects)
-      [
-        school.new(
-          "Abbey College",
-          'Long Millgate, Manchester',
-          'Primary',
-          'Independent School',
-          '£50',
-          %w{Maths English Art Physics Geography}
-        ),
-        school.new(
-          "Chetham's School of Music",
-          'Long Millgate, Manchester',
-          'Primary',
-          'Independent School',
-          '£0',
-          %w{Maths English Art Physics Music}
-        ),
-        school.new(
-          "The Creative Studio",
-          '16 Blossom Street, Manchester',
-          'Primary',
-          'Academy',
-          '£5',
-          %w{Maths Art Physics Geography}
-        )
-      ]
+class Candidates::School
+  class << self
+    def find(identifier)
+      # Note currently using id, will be converted to URN when available
+      Bookings::School.find_by(id: identifier)
```

Comment:
Should this not raise if we can't find a school? I don't think we should ever be calling this with a non existent urn

Reply:
Cheers for the review

Yeh, did debate this.

Originally I had this in the Bookings namespace before realising it was in the wrong location - didn't want to raise over there. But from within Candidates you right I should be raising.

Will change this shortly. What do you think about just letting ActiveRecord::RecordNotFound propogate or do you think it should be raising something of its own. Which could then be trapped by the controller and a 404 returned?

Reply:
I think raising AR::NotFound is best, it seems the most railsy thing for find to do and we get the 404 for free. I think 404 seems like the most sensible status to return if this find misses